### ISSUE TITLE
Undefined image error 

### ISSUE THREAD

**Issue by Tzar7** [practice404/clothes-virtual-try-on#2]
On selecting the cloth and model  and on  clicking on try it I got this error , I am selecting image from folder clothes-virtual-try-on\client-side\cloth as an input.

![pycharm](https://user-images.githubusercontent.com/109843453/201700289-9b866fbe-7b38-471a-a77b-91cf42a1f657.jpg)
![colab](https://user-images.githubusercontent.com/109843453/201700370-c329dfba-3dbe-4a8a-9b2e-6b3062208d3e.jpg)
![website appp](https://user-images.githubusercontent.com/109843453/201700650-65eb9b54-11a3-4a44-8342-9c0122752861.jpg)

**Comment by practice404** [comment]
@Tzar7 In your 2nd screenshot, error message is clear that 
> `make_dir` is not defined

Make sure you run the cell where this `make_dir` function is defined and then execute it.

**Comment by Tzar7** [comment]
> @Tzar7 In your 2nd screenshot, error message is clear that
> 
> > `make_dir` is not defined
> 
> Make sure you run the cell where this `make_dir` function is defined and then execute it.

i have done the changes , I got  same error for image 1 and image 3 but for image2  colab says data recieved
image saved, but on the website it still says undefined image error


![img4](https://user-images.githubusercontent.com/109843453/201932519-6a6c218f-3089-46aa-beb0-c4172dc39adf.png)

**Comment by practice404** [comment]
@Tzar7 make sure you read the `Setup.pdf` and followed all the steps mentioned in that.

According to this error message it seems there's something missing in the pipeline due to which the output is not generating.

I'll recommend to follow the setup.pdf thoroughly.

**Comment by zunic97** [comment]
@practice404, @Tzar7 is having expected issues. The documentation is not mentioning something.

**Comment by practice404** [comment]
@zunic97 @Tzar7 I checked the repo and made some fixes while ensuring it worked.
Please follow these steps too:
- get your own remove.bg API and replace it inside`clothes-virtual-try-on/remove_bg.py`
- get your own ngrok API and replace it inside the `C_A_MARK_2.ipynb`. Find the cell with this line and replace it with your own `!ngrok authtoken `

The rest keep as it is and it should be work, I verified it

**Comment by nhatrongbp** [comment]
so... now i have to run all the steps (cells) of “C_A_MARK_2.ipynb” file INSTEAD OF "setup.ipynb" file?

**Comment by practice404** [comment]
yes @nhatrongbp just make sure you put the correct API keys at their respective places and run the `C_A_MARK_2.ipynb` you'll get the ngrok server link

**Comment by practice404** [comment]
Let me know, if there is no problem then I can close this issue

**Comment by nhatrongbp** [comment]
2 more questions: does this project use a pre-trained model called "viton"? if it wasn't true, where can i find the training/testing step (cell)?

**Comment by nhatrongbp** [comment]
![image](https://user-images.githubusercontent.com/83824731/202678795-f3862156-84fc-439d-b0b1-7b5f07cb76c4.png)
is the name "model...jpg" and "model.._keypoints.json" wrong (the word "model" extras with 2 dots)? And why there is no image in the "cloth-mask, image-parse and openpose-img" folder? 
i think there's some issues in "run.py" or "cloth-mask.py" script, and that issue cause the "output" folder empty

**Comment by nhatrongbp** [comment]
![image](https://user-images.githubusercontent.com/83824731/202690367-93cf161e-bb43-4f0b-938b-745677391c25.png)
![image](https://user-images.githubusercontent.com/83824731/202690436-20ed3bba-a148-4d52-82e7-4ee1669e40ad.png)
and the "test_pairs.txt" file is not exist in this github repository

**Comment by nhatrongbp** [comment]
i found the problem when debugging your "run.py":
If you run line by line in the "run.py" (instead of running the whole "run.py" as the 1st screenshot below) then you will see it's not working as expected: 
your "run.py" script doesn't generate:
 - any binary masking image in /content/inputs/test/cloth-mask
 - any girl image with white background in /content/inputs/test/image
 - any segmented body image in /content/inputs/test/image-parse
 - any result in /content/inputs/test/openpose-img
 - any result in /content/inputs/test/openpose-json
=> your "run.py" does not work as expected, i think if you run line by line the "run.py" you will be able to fix these errors
*sorry for my bad english, hope you understand
![image](https://user-images.githubusercontent.com/83824731/202785191-1d707382-b179-4073-bd5a-210147652e1a.png)
![image](https://user-images.githubusercontent.com/83824731/202785754-5df6df8e-7b48-4d67-bb8f-d8c2bc734719.png)

**Comment by practice404** [comment]
Hey, @nhatrongbp I ran the script and it is working smoothly without any issues. Please check the instructions again

> @zunic97 @Tzar7 I checked the repo and made some fixes while ensuring it worked. Please follow these steps too:
> 
> * get your own remove.bg API and replace it inside`clothes-virtual-try-on/remove_bg.py`
> * get your own ngrok API and replace it inside the `C_A_MARK_2.ipynb`. Find the cell with this line and replace it with your own `!ngrok authtoken `
> 
> The rest keep as it is and it should be work, I verified it

**Comment by nhatrongbp** [comment]
![image](https://user-images.githubusercontent.com/83824731/202859170-10508ff5-c5de-4b74-857c-25f546aa8f97.png)
here the full tracebacks are:

**Comment by nhatrongbp** [comment]
hmm... i've found the pose image was generated successfully, but there's no output in the output folder
![image](https://user-images.githubusercontent.com/83824731/202859870-4cee8854-5b0a-4788-8bc7-68fccde3ad92.png)

**Comment by nhatrongbp** [comment]
i found this similar repo (screenshot below). because now im having 5 folders which contain:
- cloth image
- cloth with binary masking
- girl image
- girl image with white background
- pose keypoints json file
what will happen if i try his/her code
![image](https://user-images.githubusercontent.com/83824731/202865294-73e57912-5b63-4eab-9160-7cc1334b9f2a.png)

**Comment by practice404** [comment]
> hmm... i've found the pose image & keypoints were generated successfully, but there's no output in the output folder ![image](https://user-images.githubusercontent.com/83824731/202859870-4cee8854-5b0a-4788-8bc7-68fccde3ad92.png)

See this @nhatrongbp you are having the error of `torch not compiled with Cuda enabled` this indicates that you are using an **outdated** version of my repository. Please either **update your fork** or use my own repository 🙂
